src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -68,27 +69,54 @@ public JedisSentinelPool(String masterName, Set<String> sentinels,
   public JedisSentinelPool(String masterName, Set<String> sentinels,
       final GenericObjectPoolConfig poolConfig, int timeout, final String password,
       final int database) {
-    this(masterName, sentinels, poolConfig, timeout, timeout, password, database);
+    this(masterName, sentinels, poolConfig, timeout, timeout, null, password, database);
   }
 
+  public JedisSentinelPool(String masterName, Set<String> sentinels,
+                           final GenericObjectPoolConfig poolConfig, int timeout, final String user,
+                           final String password, final int database) {
+    this(masterName, sentinels, poolConfig, timeout, timeout, user, password, database);
+  }
+
+
   public JedisSentinelPool(String masterName, Set<String> sentinels,
       final GenericObjectPoolConfig poolConfig, int timeout, final String password,
       final int database, final String clientName) {
-    this(masterName, sentinels, poolConfig, timeout, timeout, password, database, clientName);
+    this(masterName, sentinels, poolConfig, timeout, timeout, null, password, database, clientName);
+  }
+
+  public JedisSentinelPool(String masterName, Set<String> sentinels,
+                           final GenericObjectPoolConfig poolConfig, int timeout,
+                           final String user, final String password,
+                           final int database, final String clientName) {
+    this(masterName, sentinels, poolConfig, timeout, timeout, user,  password, database, clientName);
   }
 
   public JedisSentinelPool(String masterName, Set<String> sentinels,
       final GenericObjectPoolConfig poolConfig, final int timeout, final int soTimeout,
       final String password, final int database) {
-    this(masterName, sentinels, poolConfig, timeout, soTimeout, password, database, null);
+    this(masterName, sentinels, poolConfig, timeout, soTimeout, null, password, database, null);
+  }
+
+  public JedisSentinelPool(String masterName, Set<String> sentinels,
+                           final GenericObjectPoolConfig poolConfig, final int timeout, final int soTimeout,
```

Comment:
timeout -> connectionTimeout

Reply:
Should I change the name only in this method?
Asking because if you look at other constructors, some of them are using `timeout` (mine and upper) and some other are using `connectionTimeout` (the 2 one below mine)

Reply:
@tgrall `timeout` is used when both connectionTimeout and soTimeout are equal. If you check carefully, timeout is always used alone and those two always used together.

Reply:
@sazzad16 thansk for the clarification, to be coherent, I am also updating the name of the parameter to connectionTimeout into the method that I have extended. (line 96 and line 102 )